internal/k6runner/http.go

```diff
@@ -165,7 +172,21 @@ func (r HttpRunner) request(ctx context.Context, script Script) (*RunResponse, e
 
 	req.Header.Add("content-type", "application/json")
 
-	resp, err := http.DefaultClient.Do(req)
+	// Build a tracing-enabled http client.
+	httpClient := http.Client{
+		Transport: otelhttp.NewTransport(
+			http.DefaultTransport,
+			otelhttp.WithTracerProvider(trace.SpanFromContext(ctx).TracerProvider()),
+			// Span names do not include method and path by default to avoid cardinality explosion with paths containing
+			// IDs. As this is not the case with this endpoint, we use a custom formatter that includes both.
+			otelhttp.WithSpanNameFormatter(func(_ string, r *http.Request) string {
+				return fmt.Sprintf("%s %s", r.Method, r.URL.Path)
+			}),
+			otelhttp.WithPropagators(propagation.TraceContext{}), // Send TraceIDs in outgoing requests.
+		),
+	}
```

Comment:
I distrust otel a lot...

Isn't this doing a lot of extra work over and over again?

Wouldn't it be better to create a client once (e.g. `sync.OnceValue`) 

Also, this line:

```
otelhttp.WithTracerProvider(trace.SpanFromContext(ctx).TracerProvider()),
```

isn't the tracer provider stored in the context? You don't need to extract the span to extract its provider, you can ask for the provider directly from the context. I guess there are situations where you have multiple tracer provider, but unless I'm missing something this isn't one of them.

Reply:
> Isn't this doing a lot of extra work over and over again?

Theoretically, yes. We should be able to store this client somewhere, but figuring out where was more work than just creating it anew.

Reply:
> isn't the tracer provider stored in the context? You don't need to extract the span to extract its provider, you can ask for the provider directly from the context. I guess there are situations where you have multiple tracer provider, but unless I'm missing something this isn't one of them.

Also possible. My justification is as follows: Storing the span in the context and retrieving the tracerprovider from it is "the otel way". I will do it like that everywhere I can. Storing the tracerprovider in the context directly is a hack I made up to avoid adding the tracerprovider to the constructor of every scraper and prober in the code. I will do it only where necessary, so if at some poing we want to drop the hack (the `tracing` package), we will need to change the least amount of code possible.

Reply:
> Theoretically, yes. We should be able to store this client somewhere, but figuring out where was more work than just creating it anew.

Note there's a subtle change here: when using DefaultClient, the client is not getting created over and over again. Clients _are_ reusable. If I remember things correctly, reusing clients allows the underlying code the reuse *connections*.

> My justification is as follows: Storing the span in the context and retrieving the tracerprovider from it is "the otel way" 

I know. I'm not debating that. I'm debating that the otel way doesn't buy you anything in this case. In otel it's theoretically possible to have multiple tracer providers (I challenge you to find something in the documentation that explains _why_ you might want to do that) but within the actual use case we have here, I cannot see that happening.

> To avoid adding the tracerprovider to the constructor of every scraper and prober in the code.

Hmm...

The current implementation is confusing.

- The tracerprovider is in the context, but you don't want to use it.
- Looking at the line of code I flagged, I'm not sure whether that context has or hasn't a span.

The tracerprovider should be added to the scraper and passed down from there to the point where the root span should be created, which is probably the runProber function in scraper.go. There a context is created, which is where the span should be added.

If you do that, the only way to get a tracer provider is from the span, and that resolves the issue of why it's done one way or the other.

Reply:
> The current implementation is confusing.

I will grant you that, it's not the clearest. I was hoping the comment would help, but it's true that it creates some ambiguity. Let me try to drop the context thingie and pass the tracerprovider everywhere. It will be a noisy PR.

Reply:
I've reworked the stuff. At this point it didn't make sense to keep with the `fixup!`s, so I've rebased the entire thing. Changes roughly are:
- Objects now receive the `TracerProvider` normally, not through the context. I can hardly find words for how painful it was to add this on everywhere.
- Root spans are now initiated by the scraper and the adhoc handler. Probers no longer know what a trace is. This means we lose the ability to record probe errors in a span, as Probers hide that behind the `success` boolean. We can create a child span from the Prober just with that info, but that would be duplicating it on every prober. Your call.